pkg/policy/api/egress.go

```diff
@@ -144,15 +144,27 @@ type EgressRule struct {
 	//     - 'sg-XXXXXXXXXXXXX'
 	// +optional
 	ToGroups []ToGroups `json:"toGroups,omitempty"`
+
+	aggregatedSelectors EndpointSelectorSlice
+}
+
+func (e *EgressRule) SetAggregatedSelectors() {
+	res := make(EndpointSelectorSlice, 0, len(e.ToCIDR)+len(e.ToFQDNs)+len(e.ToCIDRSet)+len(e.ToEndpoints)+len(e.ToEntities))
+	res = append(res, e.ToEndpoints...)
+	res = append(res, e.ToEntities.GetAsEndpointSelectors()...)
+	res = append(res, e.ToCIDR.GetAsEndpointSelectors()...)
+	res = append(res, e.ToCIDRSet.GetAsEndpointSelectors()...)
+	res = append(res, e.ToFQDNs.GetAsEndpointSelectors()...)
+	e.aggregatedSelectors = res
 }
 
 // GetDestinationEndpointSelectors returns a slice of endpoints selectors
 // covering all L3 destination selectors of the egress rule
 func (e *EgressRule) GetDestinationEndpointSelectors() EndpointSelectorSlice {
-	res := append(e.ToEndpoints, e.ToEntities.GetAsEndpointSelectors()...)
```

Comment:
This PR also fixes a nasty bug - if the `ToEndpoints` slice had a capacity greater than the number of values currently stored in it, and `e.ToEntities` is less than the number of remaining elements in the slice, then we would inadvertently update `ToEndpoints`! Luckily we only allocate the number of elements needed to contain the number of `ToEndpoints` from the rules provided to Cilium, so we have always re-allocated a new slice and not edited the content of the underlying `ToEndpoints` slice.

Reply:
This also applies to `FromEndpoints`.

Reply:
@ianvernon Is this actually a bug though? `e.ToEndpoints` should still refer to the portion of the memory that holds the original `e.ToEndpoints`, even if it had some spare capacity. I think it's just where in memory the elements are held - the slices themselves can refer to different portions of the same underlying set of entries. Though I'm by no means an expert in this area.. take the above with a canister of Morton.

Reply:
You are right, I was incorrect - I was moreso referring to this issue - https://medium.com/@Jarema./golang-slice-append-gotcha-e9020ff37374. 

The old implementation was correct in that it did not modify the underlying contents of the `ToEndpoints` slice.